core/src/main/java/org/elasticsearch/action/delete/DeleteResponse.java

```diff
@@ -100,4 +63,63 @@ public String toString() {
         builder.append(",shards=").append(getShardInfo());
         return builder.append("]").toString();
     }
+
+    @Override
+    public XContentBuilder innerToXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field(FOUND, result == Result.DELETED);
+        super.innerToXContent(builder, params);
+        return builder;
+    }
+
+    public static DeleteResponse fromXContent(XContentParser parser) throws IOException {
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
+
+        ParsingContext context = new ParsingContext();
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            parseXContentFields(parser, context);
+        }
+        return fromParsingContext(context);
+    }
+
+    /**
+     * Parse the current token and update the parsing context appropriately.
+     */
+    public static void parseXContentFields(XContentParser parser, ParsingContext context) throws IOException {
+        XContentParser.Token token = parser.currentToken();
+        String currentFieldName = parser.currentName();
+
+        if (FOUND.equals(currentFieldName)) {
+            if (token.isValue()) {
+                context.setFound(parser.booleanValue());
+            }
+        } else {
+            DocWriteResponse.parseInnerToXContent(parser, context);
+        }
+    }
+
+    /**
+     * Create a {@link DeleteResponse} from a parsing context.
+     */
+    public static DeleteResponse fromParsingContext(ParsingContext context) {
```

Comment:
I wonder if all these `fromParsingContext` methods could rather be `build` methods in the respective ParsingContext objects (they don't have to be called build). That way each "specialized" ParsingContext could accumulate all the info needed during the parsing and the output the right Object in the end. It would probably also enable restricting access (e.g. removing most of the getters from the ParsingContext).